### PR TITLE
Add support to open, create and close file

### PR DESCRIPTION
**Pull request by notkild**

### CODE REVIEW

**Comment by notkild** [comment]
I just noticed that open and create function can't be call on the same function, but they work if you call just one

File: std/syscall.zig [hunk 1]
```diff
@@ -268,6 +302,26 @@ pub fn write(fd: isize, buf: &const u8, count: isize) -> isize {
     syscall3(SYS_write, isize(fd), isize(buf), count)
 }
 
+pub fn open(path: []u8, flags: isize, perm: isize) -> isize {
+    syscall3(SYS_open, isize(&path[0]), flags, perm)
```
Comment (andrewrk) [inline std/syscall.zig]: path is `[]u8` which is not null terminated. Sadly I think we're going to have to do a memcpy for syscalls that require null terminated string parameters.

**Comment by andrewrk** [comment]
Thanks for the contribution. I think your problem may have to do with null terminated strings.

Also, let's get some tests going for the standard library.

**Comment by notkild** [comment]
Ok, I will try to fix that.

**Comment by andrewrk** [comment]
I'm going to visit this problem and see if we can't avoid this memcpy overhead somehow, but to get this working for now, you can do something like this:

```zig
fn open(path: []u8, flags: isize, perm: isize) -> isize {
    var buf: [path.len + 1]u8; // allocates memory on the stack
    @memcpy(&buf[0], &path[0], path.len);
    buf[path.len] = 0;
    // now you can use &buf[0] for the syscall.
}
```

**Comment by notkild** [comment]
Oh ok I understand better with this example, thanks. I will add it.

**Comment by notkild** [comment]
Done, I have commited it.

File: std/std.zig [hunk 2]
```diff
@@ -121,6 +121,16 @@ pub struct OutStream {
             }
         }
     }
+
+    pub fn close(os: &OutStream) -> %void {
+        const closed = close(os.fd);
+        if (closed < 0) {
+            return switch (-closed) {
```
Comment (andrewrk) [inline std/std.zig]: According to the man pages on my system, close can return these errors:

```
ERRORS
       EBADF  fd isn't a valid open file descriptor.

       EINTR  The close() call was interrupted by a signal; see signal(7).

       EIO    An I/O error occurred.
```

EBADF should return `error.BadFd`.

EINTR should return `error.SigInterrupt`.

the EIO and else cases look good.

**Comment by notkild** [comment]
Done, I will be more careful next time.

**Comment by andrewrk** [comment]
no worries, and thanks for the contribution! also thanks for the syntax highlighting package you made. I think that's the same format that github requires for highlighting, so that will come in handy.

**Comment by notkild** [comment]
No problem. I love this language that's why I try to contribute to it. :yum:

**Pull request merged**